iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -597,12 +600,49 @@ public void rollbackAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTab
   }
 
   @Override
-  public void preTruncateTable(org.apache.hadoop.hive.metastore.api.Table table, EnvironmentContext context)
+  public void preTruncateTable(org.apache.hadoop.hive.metastore.api.Table table, EnvironmentContext context,
+      List<String> partNames)
       throws MetaException {
     this.catalogProperties = getCatalogProperties(table);
     this.icebergTable = Catalogs.loadTable(conf, catalogProperties);
+    Map<String, PartitionField> partitionFieldMap = Maps.newHashMap();
+    for (PartitionField partField : icebergTable.spec().fields()) {
+      partitionFieldMap.put(partField.name(), partField);
+    }
+    Expression finalExp = Expressions.alwaysTrue();
+    if (partNames != null) {
+      for (String partName : partNames) {
+        String[] partColPairs = partName.split("/");
+        Expression subExp = Expressions.alwaysTrue();
+        for (String partColPair : partColPairs) {
+          String[] partColNameValue = partColPair.split("=");
```

Comment:
can you use `Warehouse.makeSpecFromName(partColPair)`?

Reply:
Not required, since we get the same string from `getPartitionsByFilter()`.

Reply:
that would do all the splits for you and no need to duplicate the logic
````
partName.split("/");
partColPair.split("=")
````

Reply:
Used `Warehouse.makeSpecFromName()`. Done.